frontend/src/components/ui/pagination.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  DotsHorizontalIcon,
+} from '@radix-ui/react-icons';
+
+import { cn } from '@/lib/utils';
+import { ButtonProps, buttonVariants } from '@/components/ui/button';
+
+const Pagination = ({ className, ...props }: React.ComponentProps<'nav'>) => (
+  <nav
+    role="navigation"
+    aria-label="pagination"
+    className={cn('mx-auto flex w-full justify-center', className)}
+    {...props}
+  />
+);
+Pagination.displayName = 'Pagination';
+
+const PaginationContent = React.forwardRef<
+  HTMLUListElement,
+  React.ComponentProps<'ul'>
+>(({ className, ...props }, ref) => (
+  <ul
+    ref={ref}
+    className={cn('flex flex-row items-center gap-1', className)}
+    {...props}
+  />
+));
+PaginationContent.displayName = 'PaginationContent';
+
+const PaginationItem = React.forwardRef<
+  HTMLLIElement,
+  React.ComponentProps<'li'>
+>(({ className, ...props }, ref) => (
+  <li ref={ref} className={cn('', className)} {...props} />
+));
+PaginationItem.displayName = 'PaginationItem';
+
+type PaginationLinkProps = {
+  isActive?: boolean;
+} & Pick<ButtonProps, 'size'> &
+  React.ComponentProps<'a'>;
+
+const PaginationLink = ({
+  className,
+  isActive,
+  size = 'icon',
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/40#issuecomment-2408165149>#dad777</a><span> - 10/11/2024, 09:57 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The PaginationLink component uses an &lt;a&gt; tag without an href attribute, which may cause accessibility issues for keyboard users.
</div>
<div id="fix">
<b>Fix:</b> Modify the PaginationLink component to use an &lt;a&gt; tag when an href is provided, and a &lt;button&gt; tag otherwise. This ensures proper semantic usage and addresses accessibility concerns.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -41,7 +41,7 @@
  type PaginationLinkProps = {
    isActive?: boolean;
  } & Pick<ButtonProps, 'size'> &
 -  React.ComponentProps<'a'>;
 +  React.ComponentProps<'a' | 'button'>;
 
  const PaginationLink = ({
    className,
 @@ -49,7 +49,7 @@
    size = 'icon',
    ...props
  }: PaginationLinkProps) => (
 -  <a
 +  {props.href ? <a
      aria-current={isActive ? 'page' : undefined}
      className={cn(
        buttonVariants({
 @@ -61,6 +61,17 @@
      {...props}
    />
 +  : <button
 +    type="button"
 +    aria-current={isActive ? 'page' : undefined}
 +    className={cn(
 +      buttonVariants({
 +        variant: isActive ? 'outline' : 'ghost',
 +        size,
 +      }),
 +      className
 +    )}
 +    {...props}
 +  />}
  );
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged